sql/core/src/test/scala/org/apache/spark/sql/internal/SQLConfSuite.scala

```diff
@@ -114,50 +114,79 @@ class SQLConfSuite extends QueryTest with SharedSQLContext {
     }
   }
 
-  test("reset - public conf") {
-    spark.sessionState.conf.clear()
-    val original = spark.conf.get(SQLConf.GROUP_BY_ORDINAL)
-    try {
-      assert(spark.conf.get(SQLConf.GROUP_BY_ORDINAL) === true)
-      sql(s"set ${SQLConf.GROUP_BY_ORDINAL.key}=false")
-      assert(spark.conf.get(SQLConf.GROUP_BY_ORDINAL) === false)
-      assert(sql(s"set").where(s"key = '${SQLConf.GROUP_BY_ORDINAL.key}'").count() == 1)
-      sql(s"reset")
-      assert(spark.conf.get(SQLConf.GROUP_BY_ORDINAL) === true)
-      assert(sql(s"set").where(s"key = '${SQLConf.GROUP_BY_ORDINAL.key}'").count() == 0)
-    } finally {
-      sql(s"set ${SQLConf.GROUP_BY_ORDINAL}=$original")
+  Seq("reset", s"reset ${SQLConf.GROUP_BY_ORDINAL.key}").foreach { resetCmd =>
+    test(s"reset - public conf $resetCmd") {
+      spark.sessionState.conf.clear()
+      val original = spark.conf.get(SQLConf.GROUP_BY_ORDINAL)
+      try {
+        assert(spark.conf.get(SQLConf.GROUP_BY_ORDINAL) === true)
+        sql(s"set ${SQLConf.GROUP_BY_ORDINAL.key}=false")
+        assert(spark.conf.get(SQLConf.GROUP_BY_ORDINAL) === false)
+        assert(sql(s"set").where(s"key = '${SQLConf.GROUP_BY_ORDINAL.key}'").count() == 1)
+        sql(resetCmd)
+        assert(spark.conf.get(SQLConf.GROUP_BY_ORDINAL) === true)
+        assert(sql(s"set").where(s"key = '${SQLConf.GROUP_BY_ORDINAL.key}'").count() == 0)
+      } finally {
+        sql(s"set ${SQLConf.GROUP_BY_ORDINAL}=$original")
```

Comment:
I think this is OK. After `clear()`, `original` gets the default value of the config, so here it wants to keep the default value unchanged.